### PR TITLE
switch opam ci to docker

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -42,14 +42,15 @@ jobs:
...
+    - run: opam install -vvv coq-core.${{ matrix.coq_version }} --ignore-constraints-on dune
+    - run: opam install -vvv coq-stdlib.${{ matrix.coq_version }} --ignore-constraints-on dune
     - run: |
-        opam pin add coq-core ${{ matrix.coq_version }}
-        opam pin add coq-stdlib ${{ matrix.coq_version }}
+        opam pin add coq-core ${{ matrix.coq_version }} --ignore-constraints-on dune
+        opam pin add coq-stdlib ${{ matrix.coq_version }} --ignore-constraints-on dune
       if: ${{ matrix.coq_version != 'dev' }}
-    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq 
+    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq  --ignore-constraints-on dune
```
Comment (gares) [inline .github/workflows/main.yml:53]:
```suggestion
    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq,dune
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -42,14 +42,15 @@ jobs:
         ocaml-compiler: ${{ matrix.ocaml_version }}
         opam-pin: false
         opam-local-packages:
+    - run: opam install dune
```
Comment (gares) [inline .github/workflows/main.yml:45]:
```suggestion
    - run: opam install dune.3.16.0
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -42,14 +42,15 @@ jobs:
...
+    - run: opam install -vvv coq-core.${{ matrix.coq_version }} --ignore-constraints-on dune
+    - run: opam install -vvv coq-stdlib.${{ matrix.coq_version }} --ignore-constraints-on dune
     - run: |
-        opam pin add coq-core ${{ matrix.coq_version }}
-        opam pin add coq-stdlib ${{ matrix.coq_version }}
+        opam pin add coq-core ${{ matrix.coq_version }} --ignore-constraints-on dune
+        opam pin add coq-stdlib ${{ matrix.coq_version }} --ignore-constraints-on dune
       if: ${{ matrix.coq_version != 'dev' }}
-    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq 
+    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq,dune
```
Comment (gares) [inline .github/workflows/main.yml:53]:
```suggestion
    - run: find / -name \*.vo
    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq,dune
```

**Comment by gares** [comment]
This is where the stdlib is:
```
...
/home/runner/work/coq-elpi/coq-elpi/_opam/lib/coq/theories/Strings/BinaryString.vo
/home/runner/work/coq-elpi/coq-elpi/_opam/lib/coq/theories/Strings/PrimStringAxioms.vo
find: ‘/proc/tty/driver’: Permission denied
/home/runner/work/coq-elpi/coq-elpi/_opam/lib/coq/user-contrib/Ltac2/Pattern.vo
/home/runner/work/coq-elpi/coq-elpi/_opam/lib/coq/user-contrib/Ltac2/Std.vo
/home/runner/work/coq-elpi/coq-elpi/_opam/lib/coq/user-contrib/Ltac2/RedFlags.vo
...
````
@proux01 is it supposed to be in coq/ (not rocq)?

**Comment by proux01** [comment]
Yes, it is still coq, this hasn't been renamed yes (and I fer we'll have a hard time doing that renaming with dune).

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -42,14 +42,16 @@ jobs:
...
+    - run: opam install -vvv coq-core.${{ matrix.coq_version }} --ignore-constraints-on dune
+    - run: opam install -vvv coq-stdlib.${{ matrix.coq_version }} --ignore-constraints-on dune
     - run: |
-        opam pin add coq-core ${{ matrix.coq_version }}
-        opam pin add coq-stdlib ${{ matrix.coq_version }}
+        opam pin add coq-core ${{ matrix.coq_version }} --ignore-constraints-on dune
+        opam pin add coq-stdlib ${{ matrix.coq_version }} --ignore-constraints-on dune
       if: ${{ matrix.coq_version != 'dev' }}
-    - run: opam install ./coq-elpi.opam --deps-only --with-test -y --ignore-constraints-on coq 
+    - run: find / -name \*.vo
```
Comment (gares) [inline .github/workflows/main.yml:53]:
```suggestion
    - run: find / -name \*.vo || true
```

**Comment by gares** [comment]
The error is now different:
```
dune build  -p coq-elpi @install
Error: path outside the workspace:
../../../_opam/lib/coq-core/plugins/ltac/ltac_plugin.cmxs from
default/theories
-> required by _build/default/theories/elpi.glob
-> required by _build/install/default/lib/coq/user-contrib/elpi/elpi.glob
-> required by _build/default/coq-elpi.install
-> required by alias install
```
I think it was a recent change to coqdep to spit recursive dependencies.
Am I right @SkySkimmer ?

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -26,12 +26,33 @@ env:
...
+        ocaml_version: ['default']
+      fail-fast: false  # don't stop jobs if one fails
+    steps:
+      - uses: actions/checkout@v3
+      - uses: coq-community/docker-coq-action@v1
+        with:
+          opam_file: 'coq-elpi.opam'
+          coq_version: ${{ matrix.coq_version }}
+          ocaml_version: ${{ matrix.ocaml_version }}
+          export: 'OPAMWITHTEST OPAMIGNORECONSTRAINTS.'  # space-separated list of variables
```
Comment (gares) [inline .github/workflows/main.yml:45]:
```suggestion
          export: 'OPAMWITHTEST OPAMIGNORECONSTRAINTS'  # space-separated list of variables
```

**Pull request merged**